packages/flutter_tools/test/general.shard/web/devices_test.dart

```diff
@@ -255,6 +255,42 @@ void main() {
     expect(processManager, hasNoRemainingExpectations);
   });
 
+  testWithoutContext('Chrome and Edge version check handles missing registry on Windows', () async {
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[]);
+    processManager.excludedExecutables.add('reg');
+
+    final Platform platform = FakePlatform(
+        operatingSystem: 'windows', environment: <String, String>{});
+    final ChromiumLauncher chromeLauncher = ChromiumLauncher(
+      fileSystem: MemoryFileSystem.test(),
+      platform: platform,
+      processManager: processManager,
+      operatingSystemUtils: null,
+      browserFinder: findChromeExecutable,
+      logger: BufferLogger.test(),
+    );
+    final MicrosoftEdgeDevice edgeDevice = MicrosoftEdgeDevice(
+      chromiumLauncher: chromeLauncher,
+      fileSystem: MemoryFileSystem.test(),
+      logger: BufferLogger.test(),
+      processManager: processManager,
+    );
+
+    expect(edgeDevice.isSupported(), true);
```

Comment:
so we will still detect edge, it's just `sdkNameAndVersion` will be empty? is this what we should be doing if we can't verify it's chromium-based?

Reply:
It looks like the empty string is used to make sure it fails the version constraint check later (although you'd think `sdkNameAndVersion` would be `'unknown` like the `GoogleChromeDevice` one):
https://github.com/flutter/flutter/blob/152af5aba7eddf8f25c9369313986e8da8973484/packages/flutter_tools/lib/src/web/web_device.dart#L282-L284

I see https://github.com/flutter/flutter/pull/56173#discussion_r419719180 as well.  \cc @jonahwilliams

Reply:
It seems like it will fall through to a version check failure, are you fine with this change as-is, @christopherfujino?  I can also dig in more and try to figure out another way to verify it's chromium based (though I don't have a Windows dev environment). I would hope `reg` being missing from the search path is pretty rare.